index.js

```diff
@@ -500,6 +520,61 @@ module.exports = class NoiseSecretStream extends Duplex {
     cb(null)
   }
 
+  _boxMessage (buffer) {
+    const MB = sodium.crypto_secretbox_MACBYTES // 16
+    const NB = sodium.crypto_secretbox_NONCEBYTES // 24
+
+    const prefix = this._boxSeq.subarray(0, 8)
+    sodium.sodium_increment(prefix)
+    if (b4a.equals(prefix, this._boxSeq.subarray(8))) throw new Error('nonce exhausted')
+
+    const secret = this._boxSecret.subarray(0, 32)
+    const envelope = b4a.allocUnsafe(8 + MB + buffer.length)
+    const nonce = envelope.subarray(0, NB)
+    const ciphertext = envelope.subarray(8)
+
+    nonce.set(this.remotePublicKey.subarray(0, NB)) // pad suffix
+    nonce.set(prefix)
+
+    sodium.crypto_secretbox_easy(ciphertext, buffer, nonce, secret)
+    return envelope
+  }
+
+  send (buffer) {
+    if (!this._boxSecret) return
+    if (!this.rawStream?.send) return // udx-stream expected
+
+    const message = this._boxMessage(buffer)
+    return this.rawStream.send(message)
+  }
+
+  trySend (buffer) {
+    if (!this._boxSecret) return
+    if (!this.rawStream?.trySend) return // udx-stream expected
+
+    const message = this._boxMessage(buffer)
```

Comment:
nit but technically this is not allowed to throw

Reply:
not throwing anymore, thanks for suggestion